torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2876,38 +2877,53 @@ at::Tensor scatter_reduce_helper(const at::Tensor& self, int64_t dim,
 
 at::Tensor XLANativeFunctions::scatter(const at::Tensor& self, int64_t dim,
                                        const at::Tensor& index,
-                                       const at::Tensor& src) {
+                                       const at::Tensor& src,
+                                       bool unique_indices) {
   XLA_FN_COUNTER("xla::");
+  XLA_CHECK(unique_indices)
+      << "scatter not have a deterministic implementation";
   return scatter_reduce_helper(self, dim, index, src, c10::nullopt);
 }
 
 at::Tensor XLANativeFunctions::scatter(const at::Tensor& self, int64_t dim,
                                        const at::Tensor& index,
-                                       const at::Scalar& value) {
+                                       const at::Scalar& value,
+                                       bool unique_indices) {
   XLA_FN_COUNTER("xla::");
+  XLA_CHECK(unique_indices)
+      << "scatter not have a deterministic implementation";
```

Comment:
Can we make it a bit more explicit, like
```
Scatter will have a nondeterministic behavior when indices are not unqiue.
```

Reply:
Good point, updated to:
> scatter will have a non-deterministic behavior when indices are not unqiue